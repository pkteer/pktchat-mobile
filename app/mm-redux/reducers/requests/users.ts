// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {combineReducers} from 'redux';

import {UserTypes} from '@mm-redux/action_types';
import {GenericAction} from '@mm-redux/types/actions';
import {UsersRequestsStatuses, RequestStatusType} from '@mm-redux/types/requests';

import {RequestStatus} from '../../constants';

import {handleRequest, initialRequestState} from './helpers';

function checkMfa(state: RequestStatusType = initialRequestState(), action: GenericAction): RequestStatusType {
    switch (action.type) {
    case UserTypes.CHECK_MFA_REQUEST:
        return {...state, status: RequestStatus.STARTED};

    case UserTypes.CHECK_MFA_SUCCESS:
        return {...state, status: RequestStatus.SUCCESS, error: null};

    case UserTypes.CHECK_MFA_FAILURE:
        return {...state, status: RequestStatus.FAILURE, error: action.error};

    default:
        return state;
    }
}

function login(state: RequestStatusType = initialRequestState(), action: GenericAction): RequestStatusType {
    switch (action.type) {
    case UserTypes.LOGIN_REQUEST:
        return {...state, status: RequestStatus.STARTED};

    case UserTypes.LOGIN_SUCCESS:
        return {...state, status: RequestStatus.SUCCESS, error: null};

    case UserTypes.LOGIN_FAILURE:
        return {...state, status: RequestStatus.FAILURE, error: action.error};

    default:
        return state;
    }
}

function autocompleteUsers(state: RequestStatusType = initialRequestState(), action: GenericAction): RequestStatusType {
    return handleRequest(
        UserTypes.AUTOCOMPLETE_USERS_REQUEST,
        UserTypes.AUTOCOMPLETE_USERS_SUCCESS,
        UserTypes.AUTOCOMPLETE_USERS_FAILURE,
        state,
        action,
    );
}

function updateMe(state: RequestStatusType = initialRequestState(), action: GenericAction): RequestStatusType {
    return handleRequest(
        UserTypes.UPDATE_ME_REQUEST,
        UserTypes.UPDATE_ME_SUCCESS,
        UserTypes.UPDATE_ME_FAILURE,
        state,
        action,
    );
}

export default (combineReducers({
    checkMfa,
    login,
    autocompleteUsers,
    updateMe,
}) as (b: UsersRequestsStatuses, a: GenericAction) => UsersRequestsStatuses);
