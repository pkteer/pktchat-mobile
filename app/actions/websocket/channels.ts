// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {fetchChannelAndMyMember} from '@actions/helpers/channels';
import {loadChannelsForTeam} from '@actions/views/channel';
import {Client4} from '@client/rest';
import {WebsocketEvents} from '@constants';
import {ChannelTypes, TeamTypes, RoleTypes} from '@mm-redux/action_types';
import {addChannelToInitialCategory} from '@mm-redux/actions/channel_categories';
import {markChannelAsRead} from '@mm-redux/actions/channels';
import {General} from '@mm-redux/constants';
import {
    getAllChannels,
    getChannel,
    getChannelsNameMapInTeam,
    getCurrentChannelId,
    getRedirectChannelNameForTeam,
} from '@mm-redux/selectors/entities/channels';
import {getConfig} from '@mm-redux/selectors/entities/general';
import {getCurrentTeamId} from '@mm-redux/selectors/entities/teams';
import {getCurrentUserId} from '@mm-redux/selectors/entities/users';
import {ActionResult, DispatchFunc, GenericAction, GetStateFunc, batchActions} from '@mm-redux/types/actions';
import {WebSocketMessage} from '@mm-redux/types/websocket';
import {getChannelByName} from '@mm-redux/utils/channel_utils';
import EventEmitter from '@mm-redux/utils/event_emitter';
import {shouldShowLegacySidebar} from '@utils/categories';

export function handleChannelConvertedEvent(msg: WebSocketMessage) {
    return (dispatch: DispatchFunc, getState: GetStateFunc): ActionResult => {
        const channelId = msg.data.channel_id;
        if (channelId) {
            const channel = getChannel(getState(), channelId);
            if (channel) {
                dispatch({
                    type: ChannelTypes.RECEIVED_CHANNEL,
                    data: {...channel, type: General.PRIVATE_CHANNEL},
                });
            }
        }
        return {data: true};
    };
}

export function handleChannelCreatedEvent(msg: WebSocketMessage) {
    return async (dispatch: DispatchFunc, getState: GetStateFunc): Promise<ActionResult> => {
        const {channel_id: channelId, team_id: teamId} = msg.data;
        const state = getState();
        const channels = getAllChannels(state);
        const currentTeamId = getCurrentTeamId(state);

        if (teamId === currentTeamId && !channels[channelId]) {
            return dispatch(fetchChannelAndAddToSidebar(msg.broadcast.channel_id, 'BATCH_WS_CHANNEL_CREATED'));
        }
        return {data: false};
    };
}

export function handleChannelDeletedEvent(msg: WebSocketMessage) {
    return (dispatch: DispatchFunc, getState: GetStateFunc): ActionResult => {
        const state = getState();
        const currentChannelId = getCurrentChannelId(state);
        const currentTeamId = getCurrentTeamId(state);
        const config = getConfig(state);
        const viewArchivedChannels = config.ExperimentalViewArchivedChannels === 'true';
        const actions: GenericAction[] = [{
            type: ChannelTypes.RECEIVED_CHANNEL_DELETED,
            data: {
                id: msg.data.channel_id,
                deleteAt: msg.data.delete_at,
                team_id: msg.broadcast.team_id,
                viewArchivedChannels,
            },
        }];

        if (msg.broadcast.team_id === currentTeamId) {
            if (msg.data.channel_id === currentChannelId && !viewArchivedChannels) {
                const channelsInTeam = getChannelsNameMapInTeam(state, currentTeamId);
                const channel = getChannelByName(channelsInTeam, getRedirectChannelNameForTeam(state, currentTeamId));
                if (channel && channel.id) {
                    actions.push({type: ChannelTypes.SELECT_CHANNEL, data: channel.id});
                }
                EventEmitter.emit(General.DEFAULT_CHANNEL, '');
            }
        }

        dispatch(batchActions(actions, 'BATCH_WS_CHANNEL_ARCHIVED'));

        return {data: true};
    };
}

export function handleChannelMemberUpdatedEvent(msg: WebSocketMessage) {
    return async (dispatch: DispatchFunc): Promise<ActionResult> => {
        try {
            const channelMember = JSON.parse(msg.data.channelMember);
            const rolesToLoad = channelMember.roles.split(' ');
            const actions: GenericAction[] = [{
                type: ChannelTypes.RECEIVED_MY_CHANNEL_MEMBER,
                data: channelMember,
            }];

            const roles = await Client4.getRolesByNames(rolesToLoad);
            if (roles.length) {
                actions.push({
                    type: RoleTypes.RECEIVED_ROLES,
                    data: roles,
                });
            }

            dispatch(batchActions(actions, 'BATCH_WS_CHANNEL_MEMBER_UPDATE'));
        } catch {
            //do nothing
        }
        return {data: true};
    };
}

export function handleChannelSchemeUpdatedEvent(msg: WebSocketMessage) {
    return async (dispatch: DispatchFunc): Promise<ActionResult> => {
        return dispatch(fetchChannelAndAddToSidebar(msg.broadcast.channel_id, 'BATCH_WS_SCHEME_UPDATE'));
    };
}

export function handleChannelUnarchiveEvent(msg: WebSocketMessage) {
    return async (dispatch: DispatchFunc, getState: GetStateFunc): Promise<ActionResult> => {
        const state = getState();
        const currentTeamId = getCurrentTeamId(state);
        const config = getConfig(state);
        const viewArchivedChannels = config.ExperimentalViewArchivedChannels === 'true';

        if (msg.broadcast.team_id === currentTeamId) {
            const actions: GenericAction[] = [{
                type: ChannelTypes.RECEIVED_CHANNEL_UNARCHIVED,
                data: {
                    id: msg.data.channel_id,
                    team_id: msg.data.team_id,
                    deleteAt: 0,
                    viewArchivedChannels,
                },
            }];

            const {data: myData}: any = await dispatch(loadChannelsForTeam(currentTeamId, true));
            if (myData?.channels && myData?.channelMembers) {
                actions.push({
                    type: ChannelTypes.RECEIVED_MY_CHANNELS_WITH_MEMBERS,
                    data: myData,
                });
            }

            dispatch(batchActions(actions, 'BATCH_WS_CHANNEL_UNARCHIVED'));
        }
        return {data: true};
    };
}

export function handleChannelUpdatedEvent(msg: WebSocketMessage) {
    return async (dispatch: DispatchFunc, getState: GetStateFunc): Promise<ActionResult> => {
        let channel;
        try {
            channel = msg.data ? JSON.parse(msg.data.channel) : null;
        } catch (err) {
            return {error: err};
        }

        const currentChannelId = getCurrentChannelId(getState());
        if (channel) {
            dispatch({
                type: ChannelTypes.RECEIVED_CHANNEL,
                data: channel,
            });

            if (currentChannelId === channel.id) {
                // Emit an event with the channel received as we need to handle
                // the changes without listening to the store
                EventEmitter.emit(WebsocketEvents.CHANNEL_UPDATED, channel);
            }
        }
        return {data: true};
    };
}

export function handleChannelViewedEvent(msg: WebSocketMessage) {
    return (dispatch: DispatchFunc, getState: GetStateFunc): ActionResult => {
        const state = getState();
        const {channel_id: channelId} = msg.data;
        const currentChannelId = getCurrentChannelId(state);
        const currentUserId = getCurrentUserId(state);

        if (channelId !== currentChannelId && currentUserId === msg.broadcast.user_id) {
            dispatch(markChannelAsRead(channelId, undefined, false));
        }
        return {data: true};
    };
}

export function handleDirectAddedEvent(msg: WebSocketMessage) {
    return async (dispatch: DispatchFunc): Promise<ActionResult> => {
        return dispatch(fetchChannelAndAddToSidebar(msg.broadcast.channel_id, 'BATCH_WS_DM_ADDED'));
    };
}

export function handleUpdateMemberRoleEvent(msg: WebSocketMessage) {
    return async (dispatch: DispatchFunc): Promise<ActionResult> => {
        const memberData = JSON.parse(msg.data.member);
        const roles = memberData.roles.split(' ');
        const actions = [];

        try {
            const newRoles = await Client4.getRolesByNames(roles);
            if (newRoles.length) {
                actions.push({
                    type: RoleTypes.RECEIVED_ROLES,
                    data: newRoles,
                });
            }
        } catch (error) {
            return {error};
        }

        actions.push({
            type: TeamTypes.RECEIVED_MY_TEAM_MEMBER,
            data: memberData,
        });

        dispatch(batchActions(actions));
        return {data: true};
    };
}

export function fetchChannelAndAddToSidebar(channelId: string, type?: string) {
    return async (dispatch: DispatchFunc, getState: GetStateFunc): Promise<ActionResult> => {
        const channelActions = await fetchChannelAndMyMember(channelId);
        let channel;

        if (channelActions.length) {
            channel = channelActions.find((el) => el.type === ChannelTypes.RECEIVED_CHANNEL);
            dispatch(batchActions(channelActions, type));
        }

        const state = getState();

        if (channel && !shouldShowLegacySidebar(state)) {
            dispatch(addChannelToInitialCategory(channel.data));
            return {data: true};
        }

        return {data: false};
    };
}
