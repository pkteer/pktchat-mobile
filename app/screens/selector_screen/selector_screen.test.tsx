// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {shallow} from 'enzyme';
import React from 'react';
import {IntlProvider} from 'react-intl';

import Preferences from '@mm-redux/constants/preferences';
import {Channel} from '@mm-redux/types/channels';
import {UserProfile} from '@mm-redux/types/users';

import SelectorScreen from './selector_screen';

const user1 = {id: 'id', username: 'username'} as UserProfile;
const user2 = {id: 'id2', username: 'username2'} as UserProfile;

const getProfiles = async () => {
    return {
        data: [user1, user2],
        error: {},
    };
};

const searchProfiles = async () => {
    return {
        data: [user2],
        error: {},
    };
};

const channel1 = {id: 'id', name: 'name', display_name: 'display_name'} as Channel;
const channel2 = {id: 'id2', name: 'name2', display_name: 'display_name2'} as Channel;

const getChannels = async () => {
    return {
        data: [channel1, channel2],
        error: {},
    };
};

const searchChannels = async () => {
    return {
        data: [channel2],
        error: {},
    };
};

const intlProvider = new IntlProvider({locale: 'en'}, {});
const {intl} = intlProvider.getChildContext();

describe('SelectorScreen', () => {
    const actions = {
        getProfiles,
        getChannels,
        searchProfiles,
        searchChannels,
    };

    const baseProps = {
        actions,
        currentTeamId: 'someId',
        onSelect: jest.fn(),
        data: [{text: 'text', value: 'value'}],
        dataSource: '',
        theme: Preferences.THEMES.denim,
    };

    beforeEach(() => {
        jest.useFakeTimers();
    });

    test('should match snapshot for explicit options', () => {
        const wrapper = shallow(
            <SelectorScreen {...baseProps}/>,
            {context: {intl}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for users', () => {
        const props = {
            ...baseProps,
            dataSource: 'users',
            data: [user1, user2],
        };

        const wrapper = shallow(
            <SelectorScreen {...props}/>,
            {context: {intl}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
        wrapper.setState({isLoading: false});
        wrapper.update();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for channels', () => {
        const props = {
            ...baseProps,
            dataSource: 'channels',
            data: [channel1, channel2],
        };

        const wrapper = shallow(
            <SelectorScreen {...props}/>,
            {context: {intl}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
        wrapper.setState({isLoading: false});
        wrapper.update();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for searching', () => {
        const props = {
            ...baseProps,
            dataSource: 'channels',
            data: [channel1, channel2],
        };

        const wrapper = shallow(
            <SelectorScreen {...props}/>,
            {context: {intl}},
        );
        wrapper.setState({isLoading: false, searching: true, term: 'name2'});
        wrapper.update();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should call getDynamicOptions if data source is dynamic', () => {
        const getDynamicOptions = jest.fn(async (term) => {
            if (term) {
                return {data: [{text: 'With Query Text', value: 'with_query'}]};
            }

            return {data: [{text: 'Without Query Text', value: 'without_query'}]};
        });

        const props = {
            ...baseProps,
            dataSource: 'dynamic',
            getDynamicOptions,
        };

        const wrapper = shallow<SelectorScreen>(
            <SelectorScreen {...props}/>,
            {context: {intl}},
        );

        jest.runAllTimers();
        setTimeout(() => {
            console.log('HERE');
            expect(props.getDynamicOptions).toHaveBeenCalledWith('');
            expect(wrapper.state().data).toEqual([
                {text: 'Without Query Text', value: 'without_query'},
            ]);
            expect(wrapper.state().searchResults).toEqual([]);

            const customList = wrapper.find('CustomList');
            expect(customList.props().data).toEqual([
                {text: 'Without Query Text', value: 'without_query'},
            ]);
        });

        // Search for value
        wrapper.instance().onSearch('mysearch');

        setTimeout(() => {
            expect(props.getDynamicOptions).toHaveBeenCalledWith('mysearch');
            expect(wrapper.state().data).toEqual([
                {text: 'Without Query Text', value: 'without_query'},
            ]);
            expect(wrapper.state().searchResults).toEqual([
                {text: 'With Query Text', value: 'with_query'},
            ]);

            const customList = wrapper.find('CustomList');
            expect(customList.props().data).toEqual([
                {text: 'With Query Text', value: 'with_query'},
            ]);
        });
    });
});
