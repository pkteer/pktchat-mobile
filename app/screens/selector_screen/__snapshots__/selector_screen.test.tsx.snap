// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectorScreen should match snapshot for channels 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "height": 38,
        "marginVertical": 5,
        "paddingLeft": 8,
      }
    }
    testID="selector.screen"
  >
    <Search
      autoCapitalize="none"
      backArrowSize={24}
      backgroundColor="transparent"
      blurOnSubmit={false}
      cancelTitle="Cancel"
      containerHeight={40}
      deleteIconSize={20}
      editable={true}
      inputHeight={33}
      inputStyle={
        Object {
          "backgroundColor": "rgba(63,67,80,0.2)",
          "color": "#3f4350",
          "fontSize": 15,
        }
      }
      keyboardAppearance="light"
      keyboardShouldPersist={false}
      keyboardType="default"
      onBlur={[Function]}
      onCancelButtonPress={[Function]}
      onChangeText={[Function]}
      onSearchButtonPress={[Function]}
      onSelectionChange={[Function]}
      placeholder="Search"
      placeholderTextColor="rgba(63,67,80,0.5)"
      returnKeyType="search"
      searchBarRightMargin={0}
      searchIconSize={24}
      showArrow={false}
      showCancel={true}
      testID="selector.search_bar"
      tintColorDelete="rgba(63,67,80,0.5)"
      tintColorSearch="rgba(63,67,80,0.5)"
      titleCancelColor="#3f4350"
      value=""
    />
  </View>
  <CustomList
    canRefresh={true}
    data={Array []}
    listType="flat"
    loading={false}
    loadingComponent={null}
    noResults={null}
    onLoadMore={[Function]}
    onRowPress={[Function]}
    renderItem={[Function]}
    shouldRenderSeparator={true}
    showNoResults={true}
    theme={
      Object {
        "awayIndicator": "#ffbc1f",
        "buttonBg": "#1c58d9",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3f4350",
        "codeTheme": "github",
        "dndIndicator": "#d24b4e",
        "errorTextColor": "#d24b4e",
        "linkColor": "#386fe5",
        "mentionBg": "#ffffff",
        "mentionColor": "#1e325c",
        "mentionHighlightBg": "#ffd470",
        "mentionHighlightLink": "#1b1d22",
        "newMessageSeparator": "#cc8f00",
        "onlineIndicator": "#3db887",
        "sidebarBg": "#1e325c",
        "sidebarHeaderBg": "#192a4d",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarTeamBarBg": "#14213e",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#5d89ea",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#28427b",
        "sidebarUnreadText": "#ffffff",
        "type": "Denim",
      }
    }
  />
</RNCSafeAreaView>
`;

exports[`SelectorScreen should match snapshot for channels 2`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "height": 38,
        "marginVertical": 5,
        "paddingLeft": 8,
      }
    }
    testID="selector.screen"
  >
    <Search
      autoCapitalize="none"
      backArrowSize={24}
      backgroundColor="transparent"
      blurOnSubmit={false}
      cancelTitle="Cancel"
      containerHeight={40}
      deleteIconSize={20}
      editable={true}
      inputHeight={33}
      inputStyle={
        Object {
          "backgroundColor": "rgba(63,67,80,0.2)",
          "color": "#3f4350",
          "fontSize": 15,
        }
      }
      keyboardAppearance="light"
      keyboardShouldPersist={false}
      keyboardType="default"
      onBlur={[Function]}
      onCancelButtonPress={[Function]}
      onChangeText={[Function]}
      onSearchButtonPress={[Function]}
      onSelectionChange={[Function]}
      placeholder="Search"
      placeholderTextColor="rgba(63,67,80,0.5)"
      returnKeyType="search"
      searchBarRightMargin={0}
      searchIconSize={24}
      showArrow={false}
      showCancel={true}
      testID="selector.search_bar"
      tintColorDelete="rgba(63,67,80,0.5)"
      tintColorSearch="rgba(63,67,80,0.5)"
      titleCancelColor="#3f4350"
      value=""
    />
  </View>
  <CustomList
    canRefresh={true}
    data={Array []}
    listType="flat"
    loading={false}
    loadingComponent={null}
    noResults={null}
    onLoadMore={[Function]}
    onRowPress={[Function]}
    renderItem={[Function]}
    shouldRenderSeparator={true}
    showNoResults={true}
    theme={
      Object {
        "awayIndicator": "#ffbc1f",
        "buttonBg": "#1c58d9",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3f4350",
        "codeTheme": "github",
        "dndIndicator": "#d24b4e",
        "errorTextColor": "#d24b4e",
        "linkColor": "#386fe5",
        "mentionBg": "#ffffff",
        "mentionColor": "#1e325c",
        "mentionHighlightBg": "#ffd470",
        "mentionHighlightLink": "#1b1d22",
        "newMessageSeparator": "#cc8f00",
        "onlineIndicator": "#3db887",
        "sidebarBg": "#1e325c",
        "sidebarHeaderBg": "#192a4d",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarTeamBarBg": "#14213e",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#5d89ea",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#28427b",
        "sidebarUnreadText": "#ffffff",
        "type": "Denim",
      }
    }
  />
</RNCSafeAreaView>
`;

exports[`SelectorScreen should match snapshot for explicit options 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "height": 38,
        "marginVertical": 5,
        "paddingLeft": 8,
      }
    }
    testID="selector.screen"
  >
    <Search
      autoCapitalize="none"
      backArrowSize={24}
      backgroundColor="transparent"
      blurOnSubmit={false}
      cancelTitle="Cancel"
      containerHeight={40}
      deleteIconSize={20}
      editable={true}
      inputHeight={33}
      inputStyle={
        Object {
          "backgroundColor": "rgba(63,67,80,0.2)",
          "color": "#3f4350",
          "fontSize": 15,
        }
      }
      keyboardAppearance="light"
      keyboardShouldPersist={false}
      keyboardType="default"
      onBlur={[Function]}
      onCancelButtonPress={[Function]}
      onChangeText={[Function]}
      onSearchButtonPress={[Function]}
      onSelectionChange={[Function]}
      placeholder="Search"
      placeholderTextColor="rgba(63,67,80,0.5)"
      returnKeyType="search"
      searchBarRightMargin={0}
      searchIconSize={24}
      showArrow={false}
      showCancel={true}
      testID="selector.search_bar"
      tintColorDelete="rgba(63,67,80,0.5)"
      tintColorSearch="rgba(63,67,80,0.5)"
      titleCancelColor="#3f4350"
      value=""
    />
  </View>
  <CustomList
    canRefresh={true}
    data={
      Array [
        Object {
          "id": "value",
          "text": "text",
          "value": "value",
        },
      ]
    }
    listType="flat"
    loading={false}
    loadingComponent={null}
    noResults={null}
    onLoadMore={[Function]}
    onRowPress={[Function]}
    renderItem={[Function]}
    shouldRenderSeparator={true}
    showNoResults={true}
    theme={
      Object {
        "awayIndicator": "#ffbc1f",
        "buttonBg": "#1c58d9",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3f4350",
        "codeTheme": "github",
        "dndIndicator": "#d24b4e",
        "errorTextColor": "#d24b4e",
        "linkColor": "#386fe5",
        "mentionBg": "#ffffff",
        "mentionColor": "#1e325c",
        "mentionHighlightBg": "#ffd470",
        "mentionHighlightLink": "#1b1d22",
        "newMessageSeparator": "#cc8f00",
        "onlineIndicator": "#3db887",
        "sidebarBg": "#1e325c",
        "sidebarHeaderBg": "#192a4d",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarTeamBarBg": "#14213e",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#5d89ea",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#28427b",
        "sidebarUnreadText": "#ffffff",
        "type": "Denim",
      }
    }
  />
</RNCSafeAreaView>
`;

exports[`SelectorScreen should match snapshot for searching 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "height": 38,
        "marginVertical": 5,
        "paddingLeft": 8,
      }
    }
    testID="selector.screen"
  >
    <Search
      autoCapitalize="none"
      backArrowSize={24}
      backgroundColor="transparent"
      blurOnSubmit={false}
      cancelTitle="Cancel"
      containerHeight={40}
      deleteIconSize={20}
      editable={true}
      inputHeight={33}
      inputStyle={
        Object {
          "backgroundColor": "rgba(63,67,80,0.2)",
          "color": "#3f4350",
          "fontSize": 15,
        }
      }
      keyboardAppearance="light"
      keyboardShouldPersist={false}
      keyboardType="default"
      onBlur={[Function]}
      onCancelButtonPress={[Function]}
      onChangeText={[Function]}
      onSearchButtonPress={[Function]}
      onSelectionChange={[Function]}
      placeholder="Search"
      placeholderTextColor="rgba(63,67,80,0.5)"
      returnKeyType="search"
      searchBarRightMargin={0}
      searchIconSize={24}
      showArrow={false}
      showCancel={true}
      testID="selector.search_bar"
      tintColorDelete="rgba(63,67,80,0.5)"
      tintColorSearch="rgba(63,67,80,0.5)"
      titleCancelColor="#3f4350"
      value="name2"
    />
  </View>
  <CustomList
    canRefresh={true}
    data={Array []}
    listType="flat"
    loading={false}
    loadingComponent={null}
    noResults={null}
    onLoadMore={[Function]}
    onRowPress={[Function]}
    renderItem={[Function]}
    shouldRenderSeparator={true}
    showNoResults={true}
    theme={
      Object {
        "awayIndicator": "#ffbc1f",
        "buttonBg": "#1c58d9",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3f4350",
        "codeTheme": "github",
        "dndIndicator": "#d24b4e",
        "errorTextColor": "#d24b4e",
        "linkColor": "#386fe5",
        "mentionBg": "#ffffff",
        "mentionColor": "#1e325c",
        "mentionHighlightBg": "#ffd470",
        "mentionHighlightLink": "#1b1d22",
        "newMessageSeparator": "#cc8f00",
        "onlineIndicator": "#3db887",
        "sidebarBg": "#1e325c",
        "sidebarHeaderBg": "#192a4d",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarTeamBarBg": "#14213e",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#5d89ea",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#28427b",
        "sidebarUnreadText": "#ffffff",
        "type": "Denim",
      }
    }
  />
</RNCSafeAreaView>
`;

exports[`SelectorScreen should match snapshot for users 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "height": 38,
        "marginVertical": 5,
        "paddingLeft": 8,
      }
    }
    testID="selector.screen"
  >
    <Search
      autoCapitalize="none"
      backArrowSize={24}
      backgroundColor="transparent"
      blurOnSubmit={false}
      cancelTitle="Cancel"
      containerHeight={40}
      deleteIconSize={20}
      editable={true}
      inputHeight={33}
      inputStyle={
        Object {
          "backgroundColor": "rgba(63,67,80,0.2)",
          "color": "#3f4350",
          "fontSize": 15,
        }
      }
      keyboardAppearance="light"
      keyboardShouldPersist={false}
      keyboardType="default"
      onBlur={[Function]}
      onCancelButtonPress={[Function]}
      onChangeText={[Function]}
      onSearchButtonPress={[Function]}
      onSelectionChange={[Function]}
      placeholder="Search"
      placeholderTextColor="rgba(63,67,80,0.5)"
      returnKeyType="search"
      searchBarRightMargin={0}
      searchIconSize={24}
      showArrow={false}
      showCancel={true}
      testID="selector.search_bar"
      tintColorDelete="rgba(63,67,80,0.5)"
      tintColorSearch="rgba(63,67,80,0.5)"
      titleCancelColor="#3f4350"
      value=""
    />
  </View>
  <CustomList
    canRefresh={true}
    data={Array []}
    listType="section"
    loading={false}
    loadingComponent={null}
    noResults={null}
    onLoadMore={[Function]}
    onRowPress={[Function]}
    renderItem={[Function]}
    shouldRenderSeparator={true}
    showNoResults={true}
    theme={
      Object {
        "awayIndicator": "#ffbc1f",
        "buttonBg": "#1c58d9",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3f4350",
        "codeTheme": "github",
        "dndIndicator": "#d24b4e",
        "errorTextColor": "#d24b4e",
        "linkColor": "#386fe5",
        "mentionBg": "#ffffff",
        "mentionColor": "#1e325c",
        "mentionHighlightBg": "#ffd470",
        "mentionHighlightLink": "#1b1d22",
        "newMessageSeparator": "#cc8f00",
        "onlineIndicator": "#3db887",
        "sidebarBg": "#1e325c",
        "sidebarHeaderBg": "#192a4d",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarTeamBarBg": "#14213e",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#5d89ea",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#28427b",
        "sidebarUnreadText": "#ffffff",
        "type": "Denim",
      }
    }
  />
</RNCSafeAreaView>
`;

exports[`SelectorScreen should match snapshot for users 2`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "height": 38,
        "marginVertical": 5,
        "paddingLeft": 8,
      }
    }
    testID="selector.screen"
  >
    <Search
      autoCapitalize="none"
      backArrowSize={24}
      backgroundColor="transparent"
      blurOnSubmit={false}
      cancelTitle="Cancel"
      containerHeight={40}
      deleteIconSize={20}
      editable={true}
      inputHeight={33}
      inputStyle={
        Object {
          "backgroundColor": "rgba(63,67,80,0.2)",
          "color": "#3f4350",
          "fontSize": 15,
        }
      }
      keyboardAppearance="light"
      keyboardShouldPersist={false}
      keyboardType="default"
      onBlur={[Function]}
      onCancelButtonPress={[Function]}
      onChangeText={[Function]}
      onSearchButtonPress={[Function]}
      onSelectionChange={[Function]}
      placeholder="Search"
      placeholderTextColor="rgba(63,67,80,0.5)"
      returnKeyType="search"
      searchBarRightMargin={0}
      searchIconSize={24}
      showArrow={false}
      showCancel={true}
      testID="selector.search_bar"
      tintColorDelete="rgba(63,67,80,0.5)"
      tintColorSearch="rgba(63,67,80,0.5)"
      titleCancelColor="#3f4350"
      value=""
    />
  </View>
  <CustomList
    canRefresh={true}
    data={Array []}
    listType="section"
    loading={false}
    loadingComponent={null}
    noResults={null}
    onLoadMore={[Function]}
    onRowPress={[Function]}
    renderItem={[Function]}
    shouldRenderSeparator={true}
    showNoResults={true}
    theme={
      Object {
        "awayIndicator": "#ffbc1f",
        "buttonBg": "#1c58d9",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3f4350",
        "codeTheme": "github",
        "dndIndicator": "#d24b4e",
        "errorTextColor": "#d24b4e",
        "linkColor": "#386fe5",
        "mentionBg": "#ffffff",
        "mentionColor": "#1e325c",
        "mentionHighlightBg": "#ffd470",
        "mentionHighlightLink": "#1b1d22",
        "newMessageSeparator": "#cc8f00",
        "onlineIndicator": "#3db887",
        "sidebarBg": "#1e325c",
        "sidebarHeaderBg": "#192a4d",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarTeamBarBg": "#14213e",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#5d89ea",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#28427b",
        "sidebarUnreadText": "#ffffff",
        "type": "Denim",
      }
    }
  />
</RNCSafeAreaView>
`;
